runners/core-construction-java/src/test/java/org/apache/beam/runners/core/construction/CommonCoderTest.java

```diff
@@ -345,6 +347,11 @@ private static Object convertValue(Object value, CommonCoder coderSpec, Coder co
       byte[] shardId = ((String) kvMap.get("shardId")).getBytes(StandardCharsets.ISO_8859_1);
       return ShardedKey.of(
           convertValue(kvMap.get("key"), coderSpec.getComponents().get(0), keyCoder), shardId);
+    } else if (s.equals(getUrn(StandardCoders.Enum.CUSTOM_WINDOW))) {
+      Map<String, Object> kvMap = (Map<String, Object>) value;
+      Instant end = new Instant(((Number) kvMap.get("end")).longValue());
+      Duration span = Duration.millis(((Number) kvMap.get("span")).longValue());
+      return new IntervalWindow(end.minus(span), span);
```

Comment:
Other places we allow any arbitrary type to be placed for the component, so how about we update this to something like:
```
Coder windowCoder = ((TimestampPrefixingWindowCoder) coder).getWindowCoder();
Object k = convertValue(kvMap.get("window"), coderSpec.getComponents().get(0), windowCoder);
```

Reply:
Done.